client/verta/verta/data_types/_verta_data_type.py

```diff
@@ -62,3 +61,10 @@ def _from_dict(cls, d):
                 return subcls._from_dict_inner(d)
 
         raise ValueError("data type {} not recognized".format(d_type))
+
+    @classmethod
+    def _from_type_string(cls, type_string):
+        for subcls in cls.__subclasses__():
+            if type_string == subcls._type_string():
+                return subcls
```

Comment:
I'd find the helpful, but optional.
```suggestion
                return subcls  # NOTE: intentionally returns the type rather than an instance
```